includes/amp-helper-functions.php

```diff
@@ -1360,6 +1350,73 @@ function amp_generate_script_hash( $script ) {
 	return 'sha384-' . $hash;
 }
 
+/**
+ * Get whether the AMP URL is overridden or not.
+ *
+ * @return bool
+ */
+function is_amp_url_overridden() {
+	return apply_filters( 'amp_override_url', false );
+}
+
+/**
+ * Get the AMP version of a URL.
+ *
+ * This simply appends the required query parameter to the URL. It does not confirm
+ * whether or not AMP is allowed on the given URL. If AMP is not supported, then a redirect
+ * to the non-AMP version will ensue.
+ *
+ * @see amp_get_permalink()
+ *
+ * @param string $url URL.
+ *
+ * @return string AMP URL.
+ */
+function amp_url( $url ) {
+	if ( is_amp_url_overridden() ) {
+		$new_url = apply_filters( 'amp_url', $url, amp_get_slug() );
+
+		if ( $new_url !== $url ) {
+			return $new_url;
+		}
+	}
+
+	return add_query_arg( amp_get_slug(), 1, $url );
+}
+
+/**
+ * Set the AMP endpoint.
+ */
+function amp_set_rewrite_rule() {
+	if ( is_amp_url_overridden() ) {
+		do_action( 'amp_set_rewrite_rule', amp_get_slug() );
+	} else {
+		add_rewrite_endpoint( amp_get_slug(), EP_PERMALINK );
+	}
+}
+
+/**
+ * Determine if the current request is for AMP or not.
+ *
+ * @return bool
+ */
+function is_amp_url() {
+	if ( is_amp_url_overridden() ) {
+		/**
+		 * Filters whether the current URL is AMP or not.
+		 *
+		 * @since 1.5.0
+		 *
+		 * @param false  $is_amp_url Whether current URL is AMP or not.
+		 * @param string $url        Current URL for the request.
+		 * @param string $amp_slug   AMP slug.
+		 */
+		return true === apply_filters( 'is_amp_url', false, amp_get_current_url(), amp_get_slug() );
+	}
+
+	return isset( $_GET[ amp_get_slug() ] ) && 1 === intval( $_GET[ amp_get_slug() ] ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
```

Comment:
I usually prefer the PHP filtering functions for dealing with `$_GET` and `$_POST`. I think this woud also remove the need for the PHPCS ignore rule.

```suggestion
	return 1 === (int) filter_input( INPUT_GET, amp_get_slug(), FILTER_SANITIZE_NUMBER_INT );
```

Reply:
Ah, TIL about `filter_input`!